### ISSUE TITLE
Project: Warehouse Inventory for Arterial Managment

### ISSUE THREAD

**Issue by dianamartin** [cityofaustin/atd-data-tech#20]
We're building a warehouse inventory management system for the Arterial Management Division to replace a paper-based process.

## Description
Arterial Management needs a warehouse management solution. They are responsible for tracking assets across multiple warehouse locations and vehicles. There are also an increasing number of non-AMD assets, such as bike racks, that are stored in the warehouse. 

![Screen Shot 2022-04-04 at 9 47 06 PM](https://user-images.githubusercontent.com/1463708/161669090-c597b653-cced-4c52-bfa1-d9e73537b8af.png)

The most pressing need is for a basic barcoding solution, however there are a range of [must-have user stories](https://docs.google.com/spreadsheets/d/1r2l_KFgjygCCp1E4oaBa5lyK2QzkVcP5qCu2u85UwcY/edit#gid=179231394) that extend into the realms of fully-featured inventory management and asset management applications. It is also critical that the software has solid API capabilities to ensure that we can build out integrations into finance and purchasing systems in the future. 

We initially looked into a warehouse inventory application offered as an add on to AIMS, but found it was [not a viable solution](https://github.com/cityofaustin/atd-data-tech/issues/64). As such, we'll be investigating both out-of-the-box vendor software along with the option of building in-house.

## Resources
 - [Comparison matrix with requirements](https://docs.google.com/spreadsheets/d/1r2l_KFgjygCCp1E4oaBa5lyK2QzkVcP5qCu2u85UwcY/edit#gid=2118741241)
- [SOP for Working Stock to be used by signal technicians](https://drive.google.com/open?id=15uozFa4yq2RWq7UJcFlt9nx3VvMgwFQz) - needs to be revisited
 - [Google Drive folder](https://drive.google.com/open?id=1Dh4UF9rNXxIrZMXorhjoPl_i0ZGUGn33)

_*City's enterprise inventory management software that Purchasing uses_

---
This Github issue represents a project of Austin Transportation's [Data & Technology Services](https://transportation.austintexas.io/about/) team. Project status is documented regularly in the comments below.

**Comment by johnclary** [comment]
@dianamartin @amenity do we have README or links to project assets somewhere on github? trying to track down a wiki entry or issue.

**Comment by johnclary** [comment]
@amenity—would you start scheduling some vendor demos? Jen gave the thumbs up.

**Comment by amenity** [comment]
@dianamartin @johnclary - Nope, my bad. I'll get them in here.

**Comment by amenity** [comment]
@johnclary added ☝️

**Comment by amenity** [comment]
**Status update:** 
After an extensive [comparison of potential solutions](https://docs.google.com/spreadsheets/d/1r2l_KFgjygCCp1E4oaBa5lyK2QzkVcP5qCu2u85UwcY/edit#gid=2118741241), we determined that building a Knack application in house is the best path forward. In the coming weeks we will resume work on an initial release that we can use to begin usability testing.

**Comment by dianamartin** [comment]
Meeting with users to review demo and learn about order management before deep dive into development of warehouse inventory. Clarifying user stories for the `Must Have` - Order Management requirements.

**Comment by amenity** [comment]
@johnclary & @dianamartin — will you stick a lil' status update here when you get a chance?

**Comment by johnclary** [comment]
**Status update** 
Application is currently under development and user testing is expected to begin as soon as the week of Nov 11th.

**Comment by dianamartin** [comment]
We had a meeting with Finance about their needs for unit cost tracking. We were able to understand what they want to see and also mock-up a unit cost calculated form.

**Comment by amenity** [comment]
@dianamartin @johnclary - can one of you write the monthly status update for this project, please? 🙏

**Comment by amenity** [comment]
Update from September: John and Chrispin will attend the 9/30 safety meeting to prep the technicians for go-live on 10/1. During this pilot phase, the paper ICF will be used in addition to the inventory module in Data Tracker. Meeting with inventory planners on 9/23 to prep them for launch.

**Comment by johnclary** [comment]
$status_update

We’re continuing to meet with technicians and inventory planners on a weekly basis to implement refinements. John met with construction services, timing engineers, and communications team to collect their work order requirements. Going live with their work order/inventory checkout form today. John also met with Ronnie to review the financial tracking workflows. A few minor issues came up and were addressed—Ronnie is now using the inventory module to update financial codes as needed. Targeting a December 1 go-live for eliminating the paper form. John to schedule more check-ins with AMD staff as we approach that date.

**Comment by johnclary** [comment]
We have continued to reconcile AIMS checkouts with Data Tracker checkouts. On the whole things are lining up nicely. We have been able to account for missing checkouts, which largely date to early December. We’ve completed the integration of task order data from eCapris, including Buyer FDUs and fund balances for backchargeable task orders. We’ve also completed an integration with the Finance & Purchasing system so that inventory items can be managed their and tied to purchase requests. We’ve also setup a process to manage working stock replenishment. Once this week’s physical inventory count is complete we will update stock levels and unit costs in the system. We’ll continue to audit AIMS vs Data Tracker checkouts until we’re comfortable with formally ending our use of AIMS. Two time-sensitive tasks remain: we need to provide Finance with the ability to tie journal voucher (JV) transactions to inventory transactions, and we need to revisit cable checkouts and make sure we’re aligned the process to manage those.

**Comment by johnclary** [comment]
Folks, an update on Data Tracker inventory based on discussion during our weekly check-in yesterday.

#### Cycle Counts
Starting next week, Johnny and Ray will begin performing weekly cycle counts on ten items at random. For now, they will keep a running list of their counts on a spreadsheet until we add Data Tracker functionality for this. Ray and Johnny—I'm asking that y'all please email the people copied on this email the results of your weekly counts.

In addition to the 10 random items, Billy has urged that cable, 1c modules, cabinets, and monitors are counted weekly, but concerns were expressed that counting cable in particular is very time-consuming. AMD folks will discuss this at their next internal inventory discussion.

#### Tablets
Ray has asked that the inventory planners receive tablets so they can stop printing out orders when they need to fill them. Billy has two extra tablets and will work with Diana to get them configured. We can purchase additional tablets if needed.

#### Supervisor approvals for inventory checkouts
We will meet on Tuesday to discuss supervisor approvals for inventory checkouts at length. In order for this to be manageable, all AMD supervisors will need to bear the responsibility for approving checkouts for their respective workgroups.

#### AIMS/Data Tracker Transition
Diana and I will meet with Van and Ronnie on Tuesday to go through the process for tracking financial data in the Data Tracker, including the process for completing Journal Vouchers (JVs).

Beyond that, we should be ready to end our use of AIMS so long as the weekly cycle counts look good. 

####  Total work order cost
I've added the total work order cost to the inventory request details page, and total transaction cost to the Issued Report (screenshot attached). Note that these costs are estimated for items which have not been issued. The unit cost is not locked in place until materials are issued. Will discuss more with Van and Ronnie on Tuesday

####  Unit cost management in Finance & Purchasing system
This is the biggest outstanding task we have—to implement dynamic unit costs based on purchase orders. For the time being, unit costs are managed in the Data Tracker through the restocking process. I do not consider this a blocker to transition out of AIMs. We should be able to complete this in the next three weeks.

**Comment by johnclary** [comment]
We're targeting an official go-live (no more AIMS) on April 1st. Recently, we've:

- Met twice to discuss cable counts. The focus has been on refining AMD business processes. No changes to the Data Tracker have been requested to accommodate cable checkouts
- Implemented the ability to document cycle counts
- Met with Ronnie and Van to review the financial tracking and reporting
- Restricted editing of inventory when a work order is closed
- Implemented the ability to re-reopen work orders, and created a new user role for lead technicians who are empowered to close work orders, but not re-open them.

#### Up next

- Discuss supervisor approvals for inventory checkouts (tomorrow, 3/12)
- Ability to manage re-order thresholds and receive alerts about low quantities on-hand
- Enable unit costs to be driven by purchase orders

**Comment by johnclary** [comment]
We've officially terminated AIMS inventory. The Data Tracker is the system of record for AMD inventory. Van will be booking the first set of financial transactions from the Data Tracker this week.

In the past month we've:
- Removed AIMS workflows from the Data Tracker
- Established new workflows for financial processing and implemented controls that ensure the integrity of transactions that have been processed.
- Reconciled unit costs between AIMS and Data Tracker

#### In Progress
- Update on-hand quantities based on 3/31 physical count
- Require supervisor approvals for inventory checkouts

#### Up next
- Ability to manage re-order thresholds and receive alerts about low quantities on-hand
- Enable unit costs to be driven by purchase orders

**Comment by amenity** [comment]
_From 6/1 AMD P&P meeting_
Ronnie is continuing to work on the reconciliation based on the early May physical count. Once Ronnie is ready, John will work with him to adjust stock levels in the Data Tracker. We've devised a new approach for the adjustments that should make it easier to audit stock levels going forward.

Completed in the past month:

-   Ability to manage reorder thresholds and monitor items that need to be re-ordered

Up next

-   Revisit cable checkout process

On Hold

-   Enable unit costs to be driven by purchase orders

-   Require supervisor approvals for inventory checkouts

For discussion:

-   Closing non-technician work orders. [Example](https://atd.knack.com/amd#work-orders-timing-engineers/view-work-orders-signal-details-timing-engineer/5fd25c097d26ba001eba6e9d/).

**Comment by johnclary** [comment]
@amenity I am closing this issue. It will be useful to retain the project label, but with the recent successful year-end count this project's scope of work is complete. Remaining issues can be treated as Data Tracker enhancements.
